Add deletion_protection field to Metastore Federation

SamanthaMathews opened the pull request:
Add deletion_protection field to make deletion actions require an explicit intent
Part of b/366209903

Part of https://github.com/hashicorp/terraform-provider-google/issues/18854

**Release Note Template for Downstream PRs (will be copied)**
```
Metastore: added `deletion_protection` field to `dataproc_metastore_service` to make deleting them require an explicit intent. `google_active_directory_domain` resources now cannot be destroyed unless `deletion_protection = false` is set for the resource.
```

c2thorn commented:
Discussing with @SamanthaMathews offline about this PR and the go rewrite migration

c2thorn committed:
Convert from Ruby to Go

migration changes (+6 squashed commits) Squashed commits: [e05eb3192] Update resource_dataproc_metastore_federation_test.go [991ee0565] migration changes [8465afe61] deletion_protection-metastore-federation [53024d449] deletion_protection-metastore-federation [788652d53] deletion_protection-metastore-federation [b0423e392] commit deletion_protection-metastore-federation

SamanthaMathews committed:
Update acceptance test

SamanthaMathews committed:
Merge branch 'GoogleCloudPlatform:main' into deletion_protection-metastore-federation
